packages/x-components/src/utils/types.ts

```diff
@@ -67,6 +67,28 @@ export type Primitive = string | number | boolean | undefined | null | symbol |
  */
 export type AnyFunction = (...args: any[]) => any;
 
+/**
+ * A function with no parameters that can return anything.
+ *
+ * @public
+ */
+export type NiladicFunction = () => any;
```

Comment:
This is the first time I read this word 😆 and Webstorm doesn't like too

![image](https://user-images.githubusercontent.com/3502075/129311979-31f627bd-483f-4401-8f51-c9c8ab691720.png)

Nevermind, everyday is a nice day to learn something new.

Reply:
Webstorm is an uneducated IDE